db/db_impl/db_impl_open.cc

```diff
@@ -1426,6 +1427,29 @@ IOStatus DBImpl::CreateWAL(uint64_t log_file_num, uint64_t recycle_log_number,
     *new_log = new log::Writer(std::move(file_writer), log_file_num,
                                immutable_db_options_.recycle_log_file_num > 0,
                                immutable_db_options_.manual_wal_flush);
+    {
+      // Track WAL in MANIFEST.
+      if (!is_db_mutex_locked) {
+        mutex_.Lock();
+      }
+
+      VersionEdit edit;
```

Comment:
If WAL or WAL dir is not synced, what do we do with this edit?

Reply:
The edit is written to manifest, but on recovery, we'll know that the wal is not synced, so we'll skip the check. We track the WAL creation here because we want to track the whole lifecycle of the WAL, instead of just tracking the syncing event.